### DISCUSSION TITLE
Help with StarkNet Contract Interaction: run_system Function Returning 'Input too short for argument

### DISCUSSION THREAD

**Discussion by otatataata** [software-mansion/starknet.py#1444] [Q&A]
Hi,

I'm working on integrating with a blockchain-based game called Influence that operates on StarkNet, a Layer 2 Ethereum scaling solution. I'm trying to call a smart contract function, `run_system`, which executes certain game actions. However, I'm running into an error that I can't seem to resolve.

### The Problem:
When I attempt to execute the function with the following `calldata`, I receive the error: "Execution failed. Failure reason: 'Input too short for arguments'."

Here is the relevant part of my code:

```
calldata = [
    4164017864394795057807229457414135500561370437549190888616,  # "ExtractResourceFinish"
    6,      
    5,      
    5,      
    7809, 
    1,      
    1,     
    4329    
]

try:
    prepared_invocation = contract.functions["run_system"].prepare_invoke_v1(name, calldata)
    # ... executing the transaction
except Exception as e:
    print(f"Error: {e}")
```

Error: `Execution failed. Failure reason: 'Input too short for arguments.`

Here is the relevant part of the ABI for the `run_system` function:

```

[
    {
        "name": "run_system",
        "type": "function",
        "inputs": [
            {
                "name": "name",
                "type": "felt252"
            },
            {
                "name": "calldata",
                "type": "array"
            }
        ],
        "outputs": [
            {
                "type": "array"
            }
        ],
        "state_mutability": "external"
    }
]
`````

Here is the definition of ExtractRessourceFinish:

```  "ExtractResourceFinish": {
    "type": "function",
    "name": "ExtractResourceFinish",
    "inputs": [
      {
        "name": "extractor",
        "type": "influence::common::types::entity::Entity"
      },
      {
        "name": "extractor_slot",
        "type": "core::integer::u64"
      },
      {
        "name": "caller_crew",
        "type": "influence::common::types::entity::Entity"
      }
    ],
    "outputs": [],
    "state_mutability": "external"
  },
```

When i copy the transaction in argent X here are the arguments: [
  "0x457874726163745265736f7572636546696e697368",
  "0x5",
  "0x5",
  "0x123c",
  "0x1",
  "0x1",
  "0xb16"
]

**Comment by kkawula** [comment]
Hi @otatataata, please check if you properly serialize the input. In the starknet.py we have a few [classes](https://starknetpy.readthedocs.io/en/latest/guide/serialization.html#data-serialization) that can help you

**Reply by kkawula** [reply]
If your input is simple, you can do it by hand for example reading this docs - [link](https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/)
